src/library/autodj/autodjfeature.cpp

```diff
@@ -277,19 +278,34 @@ void AutoDJFeature::slotCrateAutoDjChanged(int crateId, bool added) {
 
 void AutoDJFeature::slotAddRandomTrack(bool) {
 #ifdef __AUTODJCRATES__
-    // Get access to the auto-DJ playlist.
-    PlaylistDAO& playlistDao = m_pTrackCollection->getPlaylistDAO();
-    if (m_iAutoDJPlaylistId >= 0) {
-        // Get the ID of a randomly-selected track.
-        int iTrackId = m_autoDjCratesDao.getRandomTrackId();
-        if (iTrackId != -1) {
-            // Add this randomly-selected track to the auto-DJ playlist.
-            playlistDao.appendTrackToPlaylist(iTrackId, m_iAutoDJPlaylistId);
-
-            // Display the newly-added track.
-            m_pAutoDJView->onShow();
+    int retriveAttempts = 0;
+    while (retriveAttempts < 3) {
```

Comment:
this should be defined on top of this file as 
static const int kMaxRetiveAttempts = 3;